runtime/shared_common/shrinit.cpp

```diff
@@ -5082,12 +5082,13 @@ void
 j9shr_storeGCHints(J9VMThread* currentThread, UDATA heapSize1, UDATA heapSize2, BOOLEAN forceReplace)
 {
 	J9JavaVM* vm = currentThread->javaVM;
-	bool heapSizesSet = J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->localStartupHints.hintsData.flags, J9SHR_STARTUPHINTS_HEAPSIZES_SET);
+	J9SharedStartupHintsDataDescriptor* hintsData = &vm->sharedClassConfig->localStartupHints.hintsData;
+	bool heapSizesSet = J9_ARE_ALL_BITS_SET(hintsData->flags, J9SHR_STARTUPHINTS_HEAPSIZES_SET);
 
-	if (forceReplace || !heapSizesSet) {
-		vm->sharedClassConfig->localStartupHints.hintsData.heapSize1 = heapSize1;
-		vm->sharedClassConfig->localStartupHints.hintsData.heapSize2 = heapSize2;
-		vm->sharedClassConfig->localStartupHints.hintsData.flags |= J9SHR_STARTUPHINTS_HEAPSIZES_SET;
+	if (!heapSizesSet || (forceReplace && ((heapSize1 != hintsData->heapSize1) || (heapSize2 != hintsData->heapSize2)))) {
```

Comment:
This changes the meaning of the parameter `forceReplace`: the name should be changed to match the new behavior (here and where this function and pointers to it are declared).